cosmos/azure-spring-data-cosmos/cosmos-aad-sample/README.md

```diff
@@ -0,0 +1,132 @@
+# Azure Cosmos DB Spring Data RBAC with AAD Sample
+
+Azure Cosmos DB Spring Data RBAC with AAD Sample code.
+
+## Features
+
+This repo provides basic Spring Data sample code for Java SQL API to connect to [Azure Cosmos DB](https://learn.microsoft.com/azure/cosmos-db/introduction) using built-in role-based access control (RBAC), and authenticating using Azure Active Directory (AAD). See instructions below on setting up the RBAC/AAD requirements to access your Cosmos DB account and run the app successfully.
+
+## Getting Started
+
+### Prerequisites
+
+- `Java Development Kit 8` or `JDK 11` if you run the `azure-spring-data-cosmos-java-11-getting-started`. 
+- An active Azure account. If you don't have one, you can sign up for a [free account](https://azure.microsoft.com/free/). Alternatively, you can use the [Azure Cosmos DB Emulator](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator) for development and testing. As emulator https certificate is self signed, you need to import its certificate to java trusted cert store, [explained here](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator-export-ssl-certificates)
+- (Optional) SLF4J is a logging facade.
+- (Optional) [SLF4J binding](http://www.slf4j.org/manual.html) is used to associate a specific logging framework with SLF4J.
+- (Optional) Maven.
+
+SLF4J is only needed if you plan to use logging, please also download an SLF4J binding which will link the SLF4J API with the logging implementation of your choice. See the [SLF4J user manual](http://www.slf4j.org/manual.html) for more information.
+
+### Clone this repo
+
+1. git clone https://github.com/Azure-Samples/azure-spring-boot-samples.git
+2. cd cosmos/azure-spring-data-cosmos/cosmos-aad-sample
+
+### Create an Azure AD application and service principal
+
+1. Following the instructions [here](https://learn.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal) for creating an Azure AD application and service principal.
+
+1. In the [authentication](https://learn.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#authentication-two-options) section, be sure to select **option 2** to create a new application secret, and make sure you store the secret value somewhere in a text editor. 
```

Comment:
> Now we are suggesting customer to use passwordless connections. Is it possible to avoid **client-secret** here?
> 
> For example: In this sample: https://github.com/Azure-Samples/azure-spring-boot-samples/blob/spring-cloud-azure_v4.4.1/keyvault/spring-cloud-azure-starter-keyvault-secrets/property-source/src/main/resources/application.yml#L7-L11 we are using **DefaultAzureCredential** instead of **client-id** and **client-secret**.

The sample was created explicitly for customer that was using client-id and client-secret. We can DefaultAzureCredential, but I am not sure if this renders use of passwords/secrets redundant?

Reply:
> How about using [terraform ](https://www.terraform.io/)to created required resources? Just keep align with all other samples in this repository.
> 
> Example: https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_v4.4.1/keyvault/spring-cloud-azure-starter-keyvault-secrets/property-source/terraform

We can use terraform, but not all our customers use it (or want to). Still, it will be good to add this. Can we leave this for a later PR?

Reply:
@TheovanKraay Please create a new issue and paste the issue link here.